Add runtime_profile parameter to specify runtime resources

anastasop opened the pull request:
Small change to UDF exec endpoints to pass the runtime profile as described in

https://app.shortcut.com/tiledb-inc/story/15658/add-support-for-larger-udfs-including-stripe-billing-changes-for-different-pricing-of-large-udfs

Shelnutt2 commented on openapi-v1.yaml:
```diff
@@ -2059,6 +2059,13 @@ definitions:
+          The runtime profile to use for the UDF execution. Profiles define resource limits for memory and cpus.
+          If this is empty, then the UDF will execute in the standard environment of the tiledb-cloud provider.
+        x-omitempty: true
+        example: standard
```
The options should be listed out in an enum.

anastasop replied:
What about enterprise users? Shouldn't they be able to define their own names for profiles?

thetorpedodog replied:
I'm with @anastasop here that this should just be a string, and that the values should be specific to the provider. We don't want to lock in users who are hosting their own TileDB Cloud instance into only "standard" and "large". And we don't want to have to push out an API change and a client release and a server release and and and just to add a new resource profile.

johnkerl commented on openapi-v1.yaml:
```diff
@@ -2041,6 +2041,15 @@ definitions:
 
+  TileDBCloudUDFRuntimeProfile:
+    description: >
+      The runtime profile to use for the UDF execution. Profiles define resource limits for memory and cpus.
```
`cpus` -> `CPUs`

johnkerl commented on openapi-v1.yaml:
```diff
@@ -2041,6 +2041,15 @@ definitions:
+  TileDBCloudUDFRuntimeProfile:
+    description: >
+      The runtime profile to use for the UDF execution. Profiles define resource limits for memory and cpus.
+      Check the tiledb cloud provider documentation for the details.
```
`tiledb cloud` -> `TileDB Cloud`

Also it'd be great to provide a hyperlink but I understand such docs may currently be in flux ...

johnkerl commented on openapi-v1.yaml:
```diff
@@ -2041,6 +2041,15 @@ definitions:
           type: string
           x-omitempty: true
 
+  TileDBCloudUDFRuntimeProfile:
```
Will this ever apply to server-side SQL which also runs in a Docker image?

Shelnutt2 replied:
Yes

johnkerl replied:
@Shelnutt2 great! In that case @thetorpedodog maybe we could rename this. Or, accept that `UDF` sometimes means "for UDF and SQL" ... :)

thetorpedodog replied:
`ResourceClass`? `ContainerSize`? I'm fine with the imprecise name but those are a couple ideas we can take if we like them.

johnkerl replied:
oo!! I like `ResourceClass`!! Good room for future expansion there :). (`ContainerSize` is fine for now but might feel too narrow if we latter add some dimension other than size in the future ...)

thetorpedodog replied:
or `ResourceProfile`, since we use "server profile" in the UI in notebooks.

johnkerl replied:
@thetorpedodog even better! :)

anastasop replied:
We reserve `ResourceProfile` but it will be used if we decide that `runtime_profile` should be an enum. If we go with strings, it will be used in documentation

johnkerl commented:
@anastasop I'm good with this (modulo changes I already requested)

Any string-vs-enum concerns are fine; we can (IMO) deal with those in the Cloud-Py/Cloud-R API impls either way, no worries.

anastasop commented:
Changes:
1. fixed lexical issues pointed out by John
2. used `resource_class` instead of `runtime_profile`
3. added `resource_class` to SQLParameters
4. used plain strings for `resource_class`

anastasop commented:
@johnkerl can we approve this?
